cartography/models/ontology/device.py

```diff
@@ -0,0 +1,125 @@
+from dataclasses import dataclass
+
+from cartography.models.core.common import PropertyRef
+from cartography.models.core.nodes import CartographyNodeProperties
+from cartography.models.core.nodes import CartographyNodeSchema
+from cartography.models.core.nodes import ExtraNodeLabels
+from cartography.models.core.relationships import CartographyRelProperties
+from cartography.models.core.relationships import CartographyRelSchema
+from cartography.models.core.relationships import LinkDirection
+from cartography.models.core.relationships import make_target_node_matcher
+from cartography.models.core.relationships import OtherRelationships
+from cartography.models.core.relationships import TargetNodeMatcher
+
+
+@dataclass(frozen=True)
+class DeviceNodeProperties(CartographyNodeProperties):
+    id: PropertyRef = PropertyRef("hostname")
+    lastupdated: PropertyRef = PropertyRef("lastupdated", set_in_kwargs=True)
+    hostname: PropertyRef = PropertyRef("hostname", extra_index=True)
+    os: PropertyRef = PropertyRef("os")
+    os_version: PropertyRef = PropertyRef("os_version")
+    model: PropertyRef = PropertyRef("model")
+    platform: PropertyRef = PropertyRef("platform")
+    serial_number: PropertyRef = PropertyRef("serial_number", extra_index=True)
+
+
+@dataclass(frozen=True)
+class DeviceToNodeRelProperties(CartographyRelProperties):
+    lastupdated: PropertyRef = PropertyRef("lastupdated", set_in_kwargs=True)
+
+
+# (:Device)-[:HAS_OBSERVATION]->(:DuoEndpoint)
```

Comment:
I think the name `HAS_OBSERVATION` is slightly confusing

I understand that the intent is that the source record of the `(:Device)` is the `(:ModuleDevice)`. I think the reason that it feels less intuitive than, say, `(:User)-[:HAS_ACCOUNT]-(:UserAccount)` is because there is "observation" field or node anywhere. I think something like `[:OBSERVED_AS]` or even `[:SOURCE]` might be clearer. Open to ideas here though